lib/quantization/src/encoded_vectors_binary.rs

```diff
@@ -194,15 +223,88 @@ impl<TBitsStoreType: BitsStoreType, TStorage: EncodedStorage>
             encoded_vectors: storage_builder.build(),
             metadata: Metadata {
                 vector_parameters: vector_parameters.clone(),
+                encoding,
+                standard_deviations,
+                means,
             },
             bits_store_type: PhantomData,
         })
     }
 
-    fn encode_vector(vector: &[f32]) -> EncodedBinVector<TBitsStoreType> {
-        let mut encoded_vector =
-            vec![Default::default(); TBitsStoreType::get_storage_size(vector.len())];
+    fn means<'a>(
```

Comment:
Let's do struct for streaming update of statistics and move it into a dedicated file.